assets/js/modules/ads/components/common/WooCommerceRedirectModal.js

```diff
@@ -229,11 +245,25 @@ export default function WooCommerceRedirectModal( {
 							<CircularProgress size={ 14 } />
 						) : undefined
 					}
-					onClick={
-						isGoogleForWooCommerceAdsConnected
-							? getGoogleForWooCommerceRedirectURI
-							: onClose
-					}
+					onClick={ () => {
+						if (
+							( isWooCommerceActive ||
+								isGoogleForWooCommerceAdsConnected ) &&
+							! isAccountLinkedViaGoogleForWoocommerceNoticeDismissed
+						) {
+							dismissNotification(
+								'account-linked-via-google-for-woocommerce'
+							);
+						}
+						if (
+							isGoogleForWooCommerceAdsConnected ||
+							isWooCommerceActive
+						) {
+							getGoogleForWooCommerceRedirectURI();
+						} else {
+							onClose();
+						}
+					} }
```

Comment:
@10upsimon, i don't really think this should be done this way. We need to move this logic into the callback function instead of having everything in the property. We can adopt `getGoogleForWooCommerceRedirectURI` for it and rename it to a more appropriate name.